scylla-cdc-lib/src/main/java/com/scylladb/cdc/lib/CDCConsumer.java

```diff
@@ -109,11 +97,37 @@ public static Builder builder() {
         private final WorkerConfiguration.Builder workerConfigurationBuilder
                 = WorkerConfiguration.builder();
 
-        private RawChangeConsumerProvider consumerProvider;
         private int workersCount = getDefaultWorkersCount();
+        private ScheduledExecutorService executorService;
 
+        @SuppressWarnings("deprecation")
         public Builder withConsumerProvider(RawChangeConsumerProvider consumerProvider) {
-            this.consumerProvider = Preconditions.checkNotNull(consumerProvider);
+            withConsumer(consumerProvider.getForThread(0));
```

Comment:
`scylla-cdc-printer/README.md` should be updated to reflect that `RawChangeConsumerProvider` semantics changed.  Previously, you could provide a `RawChangeConsumerProvider` that would build non-thread-safe `RawChangeConsumer` for each thread. This approach is documented in that `README` file. `.getForThread(0)` makes it impossible to use it that way (I am not against it, but it should be documented).

Reply:
Done.

Reply:
I don't see README being changed in this commit?